tmt/ansible.py

```diff
@@ -82,34 +79,30 @@ class GuestAnsible(SerializableContainer):
     )
 
     @classmethod
-    def from_spec(cls, spec: Any) -> 'GuestAnsible':
+    def from_spec(cls, spec: Union[_RawGuestAnsible, None]) -> 'GuestAnsible':
         """
         Convert a YAML mapping into GuestAnsible object.
         """
         if spec is None:
             return cls()
 
-        if isinstance(spec, cls):
-            return spec
-
-        if isinstance(spec, dict):
-            return cls(**spec)  # pyright: ignore[reportUnknownArgumentType]
-
-        raise tmt.utils.SpecificationError(f"Invalid Ansible specification: {spec}")
+        return cls(
+            group=spec.get('group'),
+            vars=spec.get('vars') or {},
```

Comment:
```suggestion
            vars=spec.get('vars', {}),
```
You could also `cls(**spec)`. If it has extra keys that it would make it fail, that's something we should know.

Reply:
mypy/pyright was complaining that None case was not handled for both suggestions. I added an ignore comment